python/lsst/ts/externalscripts/maintel/setup_whitelight_flats.py

```diff
@@ -0,0 +1,158 @@
+# This file is part of ts_externalscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["SetupWhiteFlats"]
+
+
+import yaml
+from lsst.ts import salobj
+from lsst.ts.observatory.control.maintel.mtcalsys import MTCalsys
+
+
+class SetupWhiteFlats(salobj.BaseScript):
+    """Sets up Calibration Projector to perform White Light Flats
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    """
+
+    def __init__(self, index):
+        super().__init__(
+            index=index,
+            descr="Setup for White Light Flats",
+        )
+
+        self.mtcalsys = None
+        self.electrometer = None
```

Comment:
Instead of having attributes that will point to the remotes, consider making these properties, e.g.:

```
@property
def electrometer(self):
    assert self.mtcalsys is not None
    return self.mtcalsys.electrometer
```

and so on.

Reply:
I'm happy to do this, but wondering why this is better?